lib/helpers.coffee

```diff
@@ -1,33 +1,20 @@
 {Range} = require('atom')
 XRegExp = require('xregexp').XRegExp
 path = require 'path'
-child_process = require 'child_process'
+child_process = require('child_process')
 
 Helpers = module.exports =
-  shouldTriggerLinter: (linter, wasTriggeredOnChange, scopes)->
+  error: (e) ->
+    atom.notifications.addError(e.message, {detail: e.stack, dismissible: true})
+  messageKey: (message) ->
+    text = message.text || message.html
+    toReturn = message.type + '-' + message.filePath + '-' + (message.range?.serialize?()) + '-' + text
+    return toReturn.toLowerCase()
+  shouldTriggerLinter: (linter, bufferModifying, onChange, scopes) ->
     # Trigger fly linters on save, but not save linters on fly
     # Because we want to trigger onFly linters on save when the
     # user has disabled lintOnFly from config
-    return false if wasTriggeredOnChange and not linter.lintOnFly
+    return false if onChange and not linter.lintOnFly
```

Comment:
It appears you're checking whether you should lint on every change, even if the user has said they only want you to lint on save?